sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample09_ObservableEventBatch.md

```diff
@@ -0,0 +1,152 @@
+
+    // The constructor requires that sourceBatch is an empty batch so that it can track the events
+    // that are being added
+    public ObservableEventDataBatch(EventDataBatch sourceBatch)
+    {
+        _batch = sourceBatch ?? throw new ArgumentNullException(nameof(sourceBatch));
+        if (_batch.Count > 0)
+        {
+            throw new ArgumentException("The sourceBatch is not empty.", nameof(sourceBatch));
+        }
+        Events = _events.AsReadOnly();
+    }
+
+    public bool TryAdd(EventData eventData)
+    {
+        if (_batch.TryAdd(eventData))
+        {
+            _events.Add(eventData);
+            return true;
+        }
+
+        return false;
+    }
+
+    public void Dispose() => _batch.Dispose();
+
+    // Performs the needed transation to allow an ObservableEventDataBatch to be
+    // implicitly converted to an EventDataBatch
+    public static implicit operator EventDataBatch(ObservableEventDataBatch observable) => observable._batch;
+}
+```
+
+#### Benefits
+
+The benefit of this approach is that it utilizes all of the existing methods for sending and publishing events. It takes advantage of the implicit operator functionality in order to return the internal `EventDataBatch` variable, `_batch`, when these methods are called. It also is a straightfoward implementation that achieves the goal of allowing events to be seen by the application.
+
+#### Trade-offs
+
+The trade-off of this approach is that as mentioned above it does not prevent issues where the two disntinct set of events (the actual `EventBatch` events and the visible list of events) go out of sync. There are two ways that this could happen, the first is if the events returned by `ObservableDataBatch.Events` are mutated by the application. The second way this could happen is if `TryAdd` is called on the reference to the `EventDataBatch` after it has been casted from an `ObservableEventDataBatch`, this would call `EventDataBatch.TryAdd()` rather than `ObservableEventDataBatch.TryAdd()` leading to added events being in the actual batch to send that are not reflected in the visible `ObservableDataBatch.Events` variable.
+
+### Using the Observable Data Batch
+An `ObservableEventDataBatch` class is useful for any cases where an application would benefit from being able to access which events are in a batch. For example, it can be used to verify an event was added to a given batch or when dealing with failure cases, since the items in a batch that failed can be viewed and then either logged or forwarded. 
+
+### Note on Client Lifetime
+An `EventHubProducerClient` is safe to cache and use for the lifetime of the application. `CloseAsync()` is used here since this is a simple example and the client needs to be cleaned up after being used solely to demonstrate a specific behavior. In a real application, creating a new `EventHubProducerClient` on each iteration is **not** the intended use and is inefficient.  
```

Comment:
```suggestion

An `EventHubProducerClient` is safe to cache and use for the lifetime of the application. `CloseAsync` is used here since this is a simple example and the client needs to be cleaned up after being used solely to demonstrate a specific behavior. In a real application, creating a new `EventHubProducerClient` on each iteration is **not** the intended use and is inefficient.  
```